finagle-redis/src/main/scala/com/twitter/finagle/redis/Client.scala

```diff
@@ -82,19 +66,46 @@ class BaseClient(service: Service[Command, Reply]) {
     }
 
   /**
-   * Closes connection to Redis instance
+   * Select DB with specified zero-based index
+   * @param index
    */
-  def quit(): Future[Unit] =
-    doRequest(Quit) {
+  def select(index: Int): Future[Unit] =
+    doRequest(Select(index)) {
       case StatusReply(message) => Future.Unit
     }
 
   /**
-   * Select DB with specified zero-based index
-   * @param index
+   * @param host
+   * @param port
    */
-  def select(index: Int): Future[Unit] =
-    doRequest(Select(index)) {
+  def slaveOf(host: ChannelBuffer, port: Int): Future[Unit] =
+    doRequest(SlaveOf(host, StringToChannelBuffer(port.toString))) {
+      case StatusReply(message) => Future.Unit
+    }
+}
+
+/**
+ * Connects to a single Redis host
+ * @param service: Finagle service object built with the Redis codec
+ */
+class BaseClient(service: Service[Command, Reply]) {
+
+  /**
+   * Returns information and statistics about the server
+   * @param section Optional parameter can be used to select a specific section of information
+   * @return ChannelBuffer with collection of \r\n terminated lines if server has info on section
+   */
+  def info(section: ChannelBuffer = ChannelBuffers.EMPTY_BUFFER): Future[Option[ChannelBuffer]] =
+    doRequest(Info(section)) {
+      case BulkReply(message) => Future.value(Some(message))
+      case EmptyBulkReply() => Future.value(None)
```

Comment:
in what cases does it return an EmptyBulkReply?  can we just return `Future[ChannelBuffer]`?